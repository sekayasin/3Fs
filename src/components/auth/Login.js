import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import classnames from 'classnames';
import { loginUser } from '../../actions/authActions';

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      username: '',
      password: '',
      errors: {}
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();

    const user = {
      username: this.state.username,
      password: this.state.password
    };

    this.props.loginUser(user, this.props.history);
  }

  render() {
    const { errors } = this.state;

    return (
      <div>
        <section className="signin-header">
          <div className="row">
            <div className="clearfix">
              <Link to="/">
                <div className="signin-title">
                  <div className="imagebox">
                    <div className="imgShow">
                      <img
                        src={require('../../img/3FsLogo_black.png')}
                        alt="fast-food-fast logo dark"
                      />
                    </div>
                    <div className="imgHover">
                      <img
                        src={require('../../img/3FsLogo_color.png')}
                        alt="fast-food-fast logo color"
                      />
                    </div>
                  </div>
                </div>
              </Link>
              <div className="signin-signup">
                <ul className="top-navbar">
                  <li>
                    <Link to="#about-us">About us</Link>
                  </li>
                  <li>
                    <Link to="#food-delivery">Food Delivery</Link>
                  </li>
                  <li>
                    <Link to="#our-cities">Our cities</Link>
                  </li>
                  <li>
                    <Link to="#testimonials">Testimonials</Link>
                  </li>
                  <li>
                    <Link to="#blog">Blog</Link>
                  </li>
                  <li>
                    Don't have an account? <Link to="/signup">Sign up</Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </section>
        <section className="signin-form-section">
          <div className="row">
            <form
              onSubmit={this.onSubmit}
              className="signin-form"
              id="mySiginForm"
            >
              <div className="signin-form-title">
                <p>Sign In</p>
              </div>

              <div className="signin-form-subtitle">
                <p>Sign in to continue to Fast-Food-Fast</p>
              </div>

              <div className="form-elements">
                <label htmlFor="username" id="usernameerror">
                  Sign in as{' '}
                </label>
                <input
                  type="text"
                  name="username"
                  value={this.state.username}
                  onChange={this.onChange}
                  id="username"
                  placeholder="Enter your Username"
                  className={classnames({
                    'is-invalid': errors.msg
                  })}
                />
                {}
                {errors.msg && (
                  <div className="invalid-feedback">{errors.msg}</div>
                )}
              </div>

              <div className="form-elements">
                <label htmlFor="password" id="userpasserror">
                  Password{' '}
                </label>
                <input
                  type="password"
                  name="password"
                  value={this.state.password}
                  onChange={this.onChange}
                />
              </div>

              <div className="form-elements">
                <input type="submit" value="sign in" />
              </div>

              <div className="forgot-pass">
                <p>
                  <Link to="#forgot-password">Forgot password?</Link>
                </p>
              </div>
            </form>
            <div className="signin-signup-after-form">
              <p>
                Feeling hungry! Don't have an account yet?{' '}
                <Link to="/signup">Sign up here</Link>
              </p>
            </div>
          </div>
        </section>
        <section className="download-3fs-phone-app">
          <div className="row">
            <div className="fff-phone-app">
              <h4>
                Never feel hungry again! <br /> take Fast-food-fast with you!
              </h4>
              <p>Stay up-to-date on your iPhone, iPad, and Android devices.</p>
            </div>
          </div>

          <div className="row">
            <div className="fff-phone-app pos-badges">
              <div className="phone-app">
                <img
                  src={require('../../img/3fs-app-half.png')}
                  alt="3fs phone app"
                />
              </div>
              <div className="get-app-badges">
                <Link to="#download-playstore">
                  <img
                    src={require('../../img/mobile-googleplay.png')}
                    alt="download android app"
                  />
                </Link>
                <Link to="#download-applestore">
                  <img
                    src={require('../../img/mobile-applestore.png')}
                    alt="download iphone app"
                  />
                </Link>
              </div>
            </div>
          </div>
        </section>
      </div>
    );
  }
}
Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  auth: PropTypes.shape({}),
  errors: PropTypes.shape({})
};

Login.defaultProps = {
  auth: {},
  errors: {}
};

export const mapStateToProps = state => {
  return {
    auth: state.auth,
    errors: state.errors
  };
};
export default connect(
  mapStateToProps,
  { loginUser }
)(withRouter(Login));
