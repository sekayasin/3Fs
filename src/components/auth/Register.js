import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { registerUser } from '../../actions/authActions';

export class Register extends Component {
  constructor() {
    super();
    this.state = {
      first_name: '',
      last_name: '',
      username: '',
      password: '',
      email: '',
      address: '',
      tel: '',
      errors: {}
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();

    const newUser = {
      first_name: this.state.first_name,
      last_name: this.state.last_name,
      username: this.state.username,
      password: this.state.password,
      email: this.state.email,
      address: this.state.address,
      tel: this.state.tel
    };
    console.log(newUser);
    // this.props.registerUser(newUser);
    this.props.registerUser(newUser, this.props.history);
  }

  render() {
    const { errors } = this.state;

    return (
      <div>
        <section className="signin-header">
          <div className="row">
            <div className="clearfix">
              <Link to="/">
                <div className="signin-title">
                  <div className="imagebox">
                    <div className="imgShow">
                      <img
                        src={require('../../img/3FsLogo_black.png')}
                        alt="fast-food-fast logo dark"
                      />
                    </div>
                    <div className="imgHover">
                      <img
                        src={require('../../img/3FsLogo_color.png')}
                        alt="fast-food-fast logo color"
                      />
                    </div>
                  </div>
                </div>
              </Link>
              <div className="signin-signup">
                <ul className="top-navbar">
                  <li>
                    <Link to="#about-us">About us</Link>
                  </li>
                  <li>
                    <Link to="#food-delivery">Food Delivery</Link>
                  </li>
                  <li>
                    <Link to="#our-cities">Our cities</Link>
                  </li>
                  <li>
                    <Link to="#testimonials">Testimonials</Link>
                  </li>
                  <li>
                    <Link to="#blog">Blog</Link>
                  </li>
                  <li>
                    Already have an account? <Link to="/signin">Sign in</Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </section>
        <section className="signup-form-section">
          <div className="row">
            <div className="signup-intro">
              <h4>
                Feeling hungry! Sign up
                <br />
                Getting started is easy &amp; Free
              </h4>
              <p>Lucky you, you might win your self free Lunch within 48hrs</p>
            </div>
          </div>
          <div className="row">
            <form
              onSubmit={this.onSubmit}
              className="signup-form"
              id="mySignupForm"
            >
              <div className="form-elements">
                <label htmlFor="first_name">First Name</label>
                <input
                  type="text"
                  name="first_name"
                  value={this.state.first_name}
                  onChange={this.onChange}
                  placeholder="Enter your fist name"
                  required
                />
              </div>

              <div className="form-elements">
                <label htmlFor="last_name">Other name</label>
                <input
                  type="text"
                  name="last_name"
                  value={this.state.last_name}
                  onChange={this.onChange}
                  placeholder="Enter your other name"
                  required
                />
              </div>

              <div className="form-elements">
                <label htmlFor="username" id="usernametaken">
                  Username
                </label>
                <input
                  type="text"
                  name="username"
                  value={this.state.username}
                  onChange={this.onChange}
                  placeholder="Enter your preferred username"
                  required
                />
              </div>

              <div className="form-elements">
                <label htmlFor="password">Password</label>
                <input
                  type="password"
                  id="password"
                  name="password"
                  value={this.state.password}
                  onChange={this.onChange}
                  placeholder="••••••"
                  required
                />
              </div>

              <div className="form-elements">
                <label htmlFor="email" id="emailtaken">
                  Email
                </label>
                <input
                  type="email"
                  id="email"
                  name="email"
                  value={this.state.email}
                  onChange={this.onChange}
                  placeholder="Enter your email address"
                  required
                />
              </div>

              <div className="form-elements">
                <label htmlFor="address">Address</label>
                <input
                  type="text"
                  name="address"
                  value={this.state.address}
                  onChange={this.onChange}
                  placeholder="Enter your residential address"
                  required
                />
              </div>

              <div className="form-elements">
                <label htmlFor="tel">Mobile contact</label>
                <input
                  type="tel"
                  name="tel"
                  value={this.state.tel}
                  onChange={this.onChange}
                  placeholder="Enter your mobile contact"
                  required
                />
              </div>

              <div className="form-elements signup-terms-agreemt">
                <input type="checkbox" name="check_terms" required />
                <label htmlFor="check_terms">
                  Yes, I have read and consent to the{' '}
                  <Link to="#terms">Terms of Use</Link>
                  and <Link to="#agreement">Fast-food-fast Agreement.</Link>
                </label>
              </div>

              <div className="form-elements signup-terms-agreemt">
                <input type="checkbox" name="user_data_pass" required />
                <label htmlFor="user_data_pass">
                  Yes, I have read and consent to the use of my data as
                  described in
                  <Link to="#privacy">fast-food-fast's Privacy Policy</Link>
                </label>
              </div>

              <div className="form-elements signup-terms-agreemt signup-last-para">
                <p>
                  We'd love to send you a few emails on making best use of
                  fast-food-fast, special meals updates, newsletters, or
                  surveys. You can always{' '}
                  <Link to="#unsubscribe">unsubscribe.</Link>
                </p>
                <input type="radio" name="updates" defaultChecked /> Yes,
                please, i'd like to stay in touch
                <br />
                <input type="radio" name="updates" /> No, I don't want to stay
                in touch
              </div>

              <div className="form-elements signup-btn">
                <input type="submit" value="sign Up" />
              </div>
            </form>
            <div className="signup-signin-after-form">
              <p>
                Already have an account? <Link to="/signin">Sign In</Link>
              </p>
            </div>
          </div>
        </section>
      </div>
    );
  }
}
Register.propTypes = {
  registerUser: PropTypes.func.isRequired,
  auth: PropTypes.shape({}),
  errors: PropTypes.shape({})
};

Register.defaultProps = {
  auth: {},
  errors: {}
};

export const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.errors
});
export default connect(
  mapStateToProps,
  { registerUser }
)(withRouter(Register));
