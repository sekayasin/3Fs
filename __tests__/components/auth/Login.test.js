import React from 'react';
import { shallow } from 'enzyme';
import { Login, mapStateToProps } from '../../../src/components/auth/Login';

const loginUser = jest.fn();
const props = {
  onChange: jest.fn(),
  onSubmit: jest.fn(),
  loginUser
};

const state = {
  auth: '',
  errors: 'invalid'
};

describe('test login', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Login {...props} />);
  });
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('will receive props', () => {
    wrapper.setProps({ errors: 'invalid username' });
    expect(wrapper.state('errors')).toEqual('invalid username');
  });
  it('test onChange', () => {
    wrapper
      .instance()
      .onChange({ target: { name: 'username', value: 'hajat' } });
    expect(wrapper.state('username')).toEqual('hajat');
  });
  it('test onSubmit', () => {
    wrapper.instance().onSubmit({ preventDefault: jest.fn() });
    expect(loginUser).toBeCalled();
  });
  it('maps state to props', () => {
    expect(mapStateToProps(state).errors).toEqual('invalid');
  });
});
