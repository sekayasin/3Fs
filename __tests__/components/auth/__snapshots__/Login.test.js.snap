// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test login matches snapshot 1`] = `ShallowWrapper {}`;
